### PR TITLE
Use correct block levels for TRA subclasses

### PR DESCRIPTION
**Pull request by ywelsch**
Subclasses of TransportReplicationAction can currently chose to implement block levels for which the request will be blocked.

- Refresh/Flush was using the block level METADATA_WRITE although they don't operate at the cluster meta data level (but more like shard level meta data which is not represented in the block levels). Their level has been changed to `null` so that they can operate freely in the presence of blocks.
- GlobChkptSync was using WRITE although it does not make any changes to the actual documents of a shard. The level has been changed to `null` so that it can operate freely in the presence of blocks.

The PR also adds a check for closed indices in TRA so that the right exception is thrown if refresh/flush/checkpoint syncing is attempted on a closed index (before it was throwing an IndexNotFoundException, now it's throwing IndexClosedException).

An alternative to the approach taken here would be to introduce a new block level that represents writing shard "metadata" that does not change the actual documents.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java [hunk 1]
```diff
@@ -185,14 +186,14 @@ protected void resolveRequest(MetaData metaData, IndexMetaData indexMetaData, Re
      * Cluster level block to check before request execution
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: nit: add a comment about  `null` means none?

File: core/src/main/java/org/elasticsearch/action/support/replication/TransportWriteAction.java [hunk 2]
```diff
@@ -184,6 +185,16 @@ public synchronized void onSuccess(boolean forcedRefresh) {
         }
     }
 
+    @Override
+    protected ClusterBlockLevel globalBlockLevel() {
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/action/support/replication/TransportWriteAction.java]: 💯

File: core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java [hunk 3]
```diff
@@ -184,6 +187,12 @@ public void testBlocks() throws ExecutionException, InterruptedException {
         Request request = new Request();
         PlainActionFuture<Response> listener = new PlainActionFuture<>();
         ReplicationTask task = maybeTask();
+        Action action = new Action(Settings.EMPTY, "testActionWithBlocks", transportService, clusterService, shardStateAction, threadPool) {
```
Comment (bleskes) [inline core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java]: should we add a variant that checks the `null` causes blocks to be ignored?

**Review by bleskes — APPROVED** [review]
LGTM. Thanks for picking this up.

File: core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java (hunk 3 above)
Comment (ywelsch) [inline core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationActionTests.java]: sure, I've added that.

**Comment by ywelsch** [comment]
thanks @bleskes

**Pull request merged**